### PR TITLE
better loop termination condition

### LINKED ISSUE #31: loop termination condition seems wrong
Unless I am missing something, the termination condition for the loop seems incorrect:

`if l.delta.Cmp(l.prevDelta) == 0`

https://github.com/robpike/ivy/blob/master/value/loop.go#L53

`delta` is the delta between the current estimate and the previous estimate
`prevDelta` is the delta between the previous estimate and the one before that

Why should we stop if the deltas are the same? If my iterative method happens to generate estimates `0, 10, 20, 30, 33, 35, 34`, why should we stop at 30?

Seems like the correct thing to check is if `delta` is 0? I am guessing that this is how this condition ends up working in practice (it detects a string of 0s), as it's highly unlikely to actually see a string of equal but non-zero deltas.

@robpike @mjibson

### PR DESCRIPTION
**Pull request by RaduBerinde**
Replace the questionable "previous delta equals current delta" condition with a
condition that checks if delta is small (as small than the last bit of
precision).

Fixes #31.

---
This change is [Reviewable](https://reviewable.io/reviews/robpike/ivy/32)

### CODE REVIEW

File: demo/demo.out [hunk 1]
```diff
@@ -293,7 +293,7 @@ obase	10
 2302585.09299
 1
 1 0.707106781187 7.70893799599e-78 -0.707106781187 -1 -0.707106781187 2.38604460223e-76 0.707106781187 1
-6.90893484408e-77 0.69314718056 1.09861228867 1.38629436112 1.60943791243 1.79175946923
+7.77255169959e-77 0.69314718056 1.09861228867 1.38629436112 1.60943791243 1.79175946923
```
Comment (RaduBerinde) [inline demo/demo.out]: I believe this is `ln(0)`.

Comment (robpike) [inline demo/demo.out]: This is log(1)

File: value/loop.go [hunk 2]
```diff
@@ -33,44 +31,33 @@ func newLoop(conf *config.Config, name string, x *big.Float, itersPerBit uint) *
...
-			// Convergence has stopped.
-			return true
-		}
-	} else {
-		l.stallCount = 0
+	// Check if delta is no bigger than the smallest change in z that can be
+	// represented with the given precision.
+	var f big.Float
+	f.SetUint64(1)
+	f.SetMantExp(&f, z.MantExp(nil)-int(z.Prec()))
```
Comment (robpike) [inline value/loop.go]: These two lines could be one.

File: demo/demo.out (hunk 1 above)
Comment (RaduBerinde) [inline demo/demo.out]: Sorry, right, I mistyped.

File: value/loop.go (hunk 2 above)
Comment (RaduBerinde) [inline value/loop.go]: Updated. Thanks for the review!

**Pull request merged**